### PR TITLE
MAINT-28760 : search applications in the app-center is case sensitive when the app url contains uppercase letters

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**
**Problem :** search applications in the app-center is case sensitive when the app url contains uppercase letters
**Fix :**  Update search query by adding a `LOWER ()` to allow search with both uppercase and lowercase

### CODE REVIEW

**Review by azayati — DISMISSED** [review]
LGTM

**Review by rdenarie — CHANGES_REQUESTED** [review]
The change done is not enough.

The search on title or description should also be case unsensitive

File: app-center-services/src/main/java/org/exoplatform/appcenter/entity/ApplicationEntity.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
     @NamedQuery(name = "ApplicationEntity.getAppByTitle", query = "SELECT app FROM ApplicationEntity app "
         + "WHERE app.title = :title"),
     @NamedQuery(name = "ApplicationEntity.getApplicationsByKeyword", query = "SELECT app FROM ApplicationEntity app "
-        + "WHERE app.title like :title OR app.description like :description OR app.url like :url"),
+        + "WHERE app.title like :title OR app.description like :description OR LOWER(app.url) like LOWER(:url) "),
```
Comment (boubaker) [inline app-center-services/src/main/java/org/exoplatform/appcenter/entity/ApplicationEntity.java]:
```suggestion
        + "WHERE app.title like :title OR app.description like :description OR LOWER(app.url) like :url "),
```
You can do the lowercase of `:url` in DAO instead to do it in SQL part.

Comment (Rdidmohamedelhadi) [reply app-center-services/src/main/java/org/exoplatform/appcenter/entity/ApplicationEntity.java]: ok

**Review by rdenarie — DISMISSED** [review]
Lgtm

**Review by ahamdi — APPROVED** [review]

**Pull request merged**